airflow/models.py

```diff
@@ -336,7 +336,8 @@ def kill_zombies(self, session):
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(ti))
+                    ti.test_mode = False
```

Comment:
Can you provide a stack. Is this change needed as part of this PR or can you break this work into 2 PRs, with one PR just for this your changes to kill_zombies. 

I have no issue with the other parts of your PR.